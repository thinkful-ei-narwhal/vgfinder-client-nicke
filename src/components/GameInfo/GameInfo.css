.game-info-container {
  background-color: #333;
  width: 75%;
  margin: auto;
  margin-top: 20px;
  display: flex;
  justify-content: center;
  padding-top: 15px;
  padding-bottom: 45px;
}

.game-name {
  font-weight: bolder !important;
  font-size: 40px;
}

.game-info-holder {
  margin: auto;
}

.game-info-text-holder {
  color: lightgray;
  width: 100%;
  padding-left: 10%;
  padding-right: 10%;
}

@media only screen and (max-width: 520px) {
  .game-info-container {
    width: 95%;
  }
  .game-info-text-holder > p {
    font-size: 18px;
  }

  .game-name {
    font-weight: bold;
    font-size: 30px;
  }
}
