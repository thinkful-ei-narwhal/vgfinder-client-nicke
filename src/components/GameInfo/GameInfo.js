import React, { Component } from "react";
import "./GameInfo.css";
import TokenService from "./../../services/token-service";
import AddToWishlist from "../../components/AddToWishlist/AddToWishlist";
import { Button } from "./../Utils/Utils";

export default class GameInfo extends Component {
  render() {
    function myDateParse(d) {
      const date = new Date(d);
      const options = {
        weekday: "long",
        year: "numeric",
        month: "long",
        day: "numeric",
      };
      return date.toLocaleDateString(window.navigator.languages[0], options);
    }
    const {
      description,
      genre,
      rating,
      release_date,
      developer,
    } = this.props.game;

    return (
      <div className="game-info-container">
        <div className="game-info-text-holder game-info-holder">
          <p>{description}</p>
          <p>Genre: {genre}</p>
          <p>Rating: {rating}</p>
          <p>Release Date: {myDateParse(release_date)}</p>
          <p>Developer: {developer}</p>
          {TokenService.hasAuthToken() ? (
            <AddToWishlist game={this.props.game} />
          ) : (
            <Button disabled={true}>Sign in to add game to wishlist</Button>
          )}
        </div>
      </div>
    );
  }
}
