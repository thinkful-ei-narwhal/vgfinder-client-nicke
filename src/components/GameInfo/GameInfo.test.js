import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GameInfo from "./GameInfo";
import stubGameData from "./../../../testhelpers/GameFixtures";

describe(`GameInfo component`, () => {
  it("renders a GameInfo with supplied props", () => {
    const wrapper = shallow(<GameInfo game={stubGameData} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
