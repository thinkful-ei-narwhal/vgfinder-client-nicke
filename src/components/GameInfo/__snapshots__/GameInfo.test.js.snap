// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo component renders a GameInfo with supplied props 1`] = `
<Fragment>
  <p />
  <p>
    Genre: 
  </p>
  <p>
    Rating: 
  </p>
  <p>
    Release Date: 
    Invalid Date
  </p>
  <p>
    Developer: 
  </p>
</Fragment>
`;
